libraries/Microsoft.Bot.Builder.Runtime/Providers/Channel/DeclarativeChannelProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using AdaptiveExpressions.Properties;
+using Microsoft.Bot.Builder.Runtime.Extensions;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using IChannel = Microsoft.Bot.Connector.Authentication.IChannelProvider;
+
+namespace Microsoft.Bot.Builder.Runtime.Providers.Channel
+{
+    [JsonObject]
+    public class DeclarativeChannelProvider : IChannelProvider
```

Comment:
I'm not sure this is a step forward. 

@johnataylor this expands on IChannelProvider... is that a direction the SDK is headed?

Reply:
This is really just to allow it to be declared in settings.  At the point the new auth is ready, we'll adjust to use that.